d3a_api_client/utils.py

```diff
@@ -219,4 +219,24 @@ def execute_function_util(function: callable, function_name):
         function()
     except Exception as e:
         logger.error(
-            f"{function_name} raised exception: {str(e)}. \n Traceback: {str(traceback.format_exc())}")
\ No newline at end of file
+            f"{function_name} raised exception: {str(e)}. \n Traceback: {str(traceback.format_exc())}")
+
+
+def log_market_progression(message):
+    try:
+        event = message.get("event", None)
+        if event not in ["tick", "market"]:
+            return
+        headers = ["event", ]
+        table_data = [event, ]
+        data_dict = message.get("content")[0] if "content" in message.keys() else message
+        if "slot_completion" in data_dict:
+            headers.append("slot_completion")
+            table_data.append(data_dict.get("slot_completion"))
+        if "start_time" in data_dict.keys():
```

Comment:
```suggestion
        if "start_time" in data_dict:
```
sorry, one more like https://github.com/gridsingularity/d3a-api-client/pull/90#discussion_r569243263